components/PlaybackTask.brs

```diff
@@ -1,253 +1,348 @@
+    m.skipAds = false   ' Flag to skip non-truex ads
+    m.lastPosition = 0  ' Tracks last position, primarily for seeking purposes
+    m.videoPlayer = invalid ' Hold reference to player component from render thread
+    m.raf = invalid
+    m.currentAdPod = invalid ' Current ad pod in use or processing
 end sub
 
+'-------------------------------------------
+' Initialize video player
+'-----------------------------------------
 sub setupVideo()
-  ? "TRUE[X] >>> setupVideo()"
+    ? "TRUE[X] >>> setupVideo()"
 
-  videoPlayer = m.top.video
-  videoContent = createObject("roSGNode", "ContentNode")
+    videoPlayer = m.top.video
+    videoContent = createObject("roSGNode", "ContentNode")
 
-  videoContent.url = "http://ctv.truex.com.s3.amazonaws.com/assets/reference-app-stream-no-ads-720p.mp4"
-  videoContent.length = 22 * 60
+    videoContent.url = "http://ctv.truex.com/assets/reference-app-stream-no-ads-720p.mp4"
+    videoContent.length = 22 * 60
 
-  videoContent.title = "video title"
-  videoContent.streamFormat = "mp4"
-  videoContent.playStart = 0
+    videoContent.title = "The true[X] Employee Experience"
+    videoContent.streamFormat = "mp4"
+    videoContent.playStart = 0
 
-  videoPlayer.content = videoContent
-  videoPlayer.SetFocus(true)
-  videoPlayer.visible = true
-  videoPlayer.observeField("position", m.port)
-  videoPlayer.EnableCookies()
+    videoPlayer.content = videoContent
+    videoPlayer.SetFocus(true)
+    videoPlayer.visible = true
+    videoPlayer.observeField("position", m.port)
+    videoPlayer.EnableCookies()
 
-  m.videoPlayer = videoPlayer
+    m.videoPlayer = videoPlayer
 end sub
 
+'-------------------------------------------
+' Initialize Roku Ads Framework (RAF)
+' Also setup ads via RAF if known ahead of time
+'-----------------------------------------
 sub setupRaf()
-  ? "TRUE[X] >>> setupRaf()"
-  raf = Roku_Ads()
-  raf.enableAdMeasurements(true)
-  raf.setContentGenre("Entertainment")
-  raf.setContentId("TrueXSample")
-
-  raf.SetDebugOutput(false) 'debugging
-  raf.SetAdPrefs(false)
-
-  adUrl = m.top.adUrl
-  if adUrl = invalid OR adUrl = ""
-    ' adUrl = "pkg:/res/adpods/vast-regular.xml"  ' Regular Ads
-    adUrl = "pkg:/res/adpods/vmap-truex.xml" ' Preroll + Midroll Truex experience
-    ' adUrl = "pkg:/res/adpod/truex-pod-preroll.xml"
-    ' adUrl = "pkg:/res/adpod/truex-pod-skip.xml" ' Forced standalone skip pod.  Can be inserted in a full ad payload
-  end if 
-  raf.setAdUrl(adUrl)
-
-  m.raf = raf
+    ? "TRUE[X] >>> setupRaf()"
+    raf = Roku_Ads()
+    raf.enableAdMeasurements(true)
+    raf.setContentGenre("Entertainment")
+    raf.setContentId("TrueXSample")
+
+    raf.SetDebugOutput(true) 'debugging
+    raf.SetAdPrefs(false)
+
+    adUrl = m.top.adUrl
+    if adUrl = invalid OR adUrl = ""
+        adUrl = "pkg:/res/adpods/vmap-truex.xml" ' Preroll + Midroll Truex experience
+        ' adUrl = "pkg:/res/adpod/truex-pod-preroll.xml"  ' Can check individual vast pods
```

Comment:
Cool, but if you are going to leave this line in here, I assume we tested this alternate xml file and it works? 😅

Reply:
Yep :).  The caveat is Roku always treats it as a preroll, which makes sense since you didn't provide contextual time information.